### PR TITLE
fix: API change

### LINKED ISSUE #19739: [Task]: Refactor List Widget V2 Api
1. Add `triggereitem` to List v2
2. Rename `onRowClick` to `onItemClick` on List v2

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description
Add triggereitem to List v2
Rename onRowClick to onItemClick on List v2

Fixes #19739

## Type of change

- This change requires a documentation update

## How Has This Been Tested?
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/widgets/ListWidgetV2/widget/propertyConfig.ts [hunk 1]
```diff
@@ -197,9 +197,9 @@ export const PropertyPaneContentConfig = [
     sectionName: "Events",
     children: [
       {
-        propertyName: "onRowClick",
-        helpText: "Triggers an action when a row is clicked",
-        label: "onRowClick",
+        propertyName: "onItemClick",
+        helpText: "Triggers an action when an item is clicked",
```
Comment (Tooluloope) [inline app/client/src/widgets/ListWidgetV2/widget/propertyConfig.ts:201]: `Triggers an action when an item is clicked` 
@ashit-rath would this be clear to the user?

File: app/client/src/widgets/ListWidgetV2/widget/derived.js [hunk 2]
```diff
@@ -20,6 +20,26 @@ export default {
     return selectedItem;
   },
   //
+  getTriggeredItem: (props, moment, _) => {
+    const triggeredItemIndex =
+      props.triggeredItemIndex === undefined ||
+      Number.isNaN(parseInt(props.triggeredItemIndex))
+        ? -1
+        : parseInt(props.triggeredItemIndex);
```
Comment (ashit-rath) [inline app/client/src/widgets/ListWidgetV2/widget/derived.js:28]: In what cases do we expect the triggeredItemIndex to be a non-integer?

File: app/client/src/widgets/ListWidgetV2/widget/derived.js [hunk 3]
```diff
@@ -20,6 +20,26 @@ export default {
...
+    const items = props.listData || [];
+
+    if (triggeredItemIndex === -1) {
+      const emptyRow = { ...items[0] };
+      Object.keys(emptyRow).forEach((key) => {
+        emptyRow[key] = "";
```
Comment (ashit-rath) [inline app/client/src/widgets/ListWidgetV2/widget/derived.js:34]: Do we need to show it in this way? Can't we return `null`?

File: app/client/src/widgets/ListWidgetV2/widget/propertyConfig.ts (hunk 1 above)
Comment (ashit-rath) [reply app/client/src/widgets/ListWidgetV2/widget/propertyConfig.ts:201]: We will be getting all of theses reviewed by someone. 
cc: @dilippitchika

File: app/client/src/widgets/ListWidgetV2/widget/derived.js (hunk 2 above)
Comment (Tooluloope) [reply app/client/src/widgets/ListWidgetV2/widget/derived.js:28]: Probably at the initial launch of the List widget.

Remember the issue we had when the Widget is a `Skeleton Widget` and the default meta Properties were undefined.

File: app/client/src/widgets/ListWidgetV2/widget/derived.js (hunk 3 above)
Comment (Tooluloope) [reply app/client/src/widgets/ListWidgetV2/widget/derived.js:34]: This is carried over from V1.
That's why I left it for both SelectedItem and triggeredItem

Comment (ashit-rath) [reply app/client/src/widgets/ListWidgetV2/widget/derived.js:34]: Can you check with Dilip once?

Comment (Tooluloope) [reply app/client/src/widgets/ListWidgetV2/widget/derived.js:34]: Okay, when he's back.
I think he's currently AFK

File: app/client/src/widgets/ListWidgetV2/widget/derived.js (hunk 2 above)
Comment (ashit-rath) [reply app/client/src/widgets/ListWidgetV2/widget/derived.js:28]: Apart from `undefined` are we getting any other value?

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**